src/java/org/apache/cassandra/locator/ReplicaPlan.java

```diff
@@ -92,14 +98,14 @@ public String toString()
 
     public static class ForTokenRead extends ForRead<EndpointsForToken>
     {
-        public ForTokenRead(Keyspace keyspace, ConsistencyLevel consistencyLevel, EndpointsForToken candidates, EndpointsForToken contact)
+        public ForTokenRead(Keyspace keyspace, AbstractReplicationStrategy replicationStrategy, ConsistencyLevel consistencyLevel, EndpointsForToken candidates, EndpointsForToken contacts)
```

Comment:
Nit: the list of arguments in the constructors is starting to be quite long, perhaps we could break it
```suggestion
        public ForTokenRead(Keyspace keyspace,
                            AbstractReplicationStrategy replicationStrategy,
                            ConsistencyLevel consistencyLevel,
                            EndpointsForToken candidates,
                            EndpointsForToken contacts)
```